addons/sale_product_configurator/controllers/main.py

```diff
@@ -48,110 +45,32 @@ def get_product_configurator_values(
         if not combination:
             combination = product_template._get_first_possible_combination()
```

Comment:
Do we really need this duplicate part? If it's only for optionalProduct, couldn't it be done in `_get_optional_product`?

Reply:
I guess we could move it to the ``if`` statement you mentioned here => https://github.com/odoo/odoo/pull/129118#discussion_r1357140923